linera-core/src/local_node.rs

```diff
@@ -729,11 +729,18 @@ where
         node: &impl LocalValidatorNode,
         blob_id: BlobId,
     ) -> Option<HashedBlob> {
-        match node.download_blob(blob_id).await.map(Blob::into_hashed) {
-            Ok(hashed_blob) if hashed_blob.id() == blob_id => Some(hashed_blob),
-            Ok(_) => {
-                tracing::info!("Validator {name} sent an invalid blob {blob_id}.");
-                None
+        match node.download_blob(blob_id).await {
+            Ok(blob) => {
+                let hashed_blob = match blob_id.blob_type {
+                    BlobType::Data => blob.with_data_blob_id(),
+                };
+
+                if hashed_blob.id() == blob_id {
+                    Some(hashed_blob)
+                } else {
```

Comment:
We could have a method `blob.with_id(blob_id)` that does the check (i.e. returns an option).